fe/fe-core/src/main/java/com/starrocks/qe/scheduler/DefaultWorkerProvider.java

```diff
@@ -141,6 +141,23 @@ public DefaultWorkerProvider(ImmutableMap<Long, ComputeNode> id2Backend,
         this.preferComputeNode = preferComputeNode;
     }
 
+    @VisibleForTesting
+    public DefaultWorkerProvider(
+            ImmutableMap<Long, ComputeNode> id2ComputeNode,
+            ImmutableMap<Long, ComputeNode> availableID2ComputeNode) {
```

Comment:
It seems this is equals to `DefaultWorkerProvider(ImmutableMap.of(), ImmutableMap.of(), id2ComputeNode, availableID2ComputeNode)`? Maybe we needn't expose this new constructor.

Reply:
> It seems this is equals to `DefaultWorkerProvider(ImmutableMap.of(), ImmutableMap.of(), id2ComputeNode, availableID2ComputeNode)`? Maybe we needn't expose this new constructor.

I expose this new constructor because if availableID2ComputeNode is empty,  the default constuctor "public DefaultWorkerProvider(ImmutableMap<Long, ComputeNode> id2Backend,
                                 ImmutableMap<Long, ComputeNode> id2ComputeNode,
                                 ImmutableMap<Long, ComputeNode> availableID2Backend,
                                 ImmutableMap<Long, ComputeNode> availableID2ComputeNode,
                                 boolean preferComputeNode)" will set usedComputeNode = false. So selectNextWorker() will never  report: "Compute node not found:"